docs_src/design/adr/core/0022-UoM.md

```diff
@@ -0,0 +1,199 @@
+
+## Decision
+
+It would be speculative and inappropriate for EdgeX to select a unit of measure standard which is not widely adopted in the industry or choose a static unit of measure list that is incomplete with regard to possible IoT / edge use case needs.  At this time, there does not appear to be a single and unequivocal standard for units of measure that encompasses all EdgeX related use cases (now and in the future). 
+
+Therefore, EdgeX chooses not to select or adopt a unit of measure specification, standard, or code list to apply across the platform.  Instead, EdgeX adopters will be allowed to optionally specify which unit of measure specification, standard, or unit of measure code list they would like used in their instance(s) of EdgeX.
+
+### Specifying the Units of Measure
+
+Units of measure allowed by the instance of EdgeX will be specified in a configuration file (in TOML format).  This can either be specified directly in the core metadata configuration file (but then hard to change when using a containerized or snap form of the application) or in a separate and external configuration TOML file (making it easier for an adopter to provide units of measure even when the application is containerized or snapped up).  
+
+The units of measure in the configuration file can be attributed, optionally, to a specification, document, or other UoM definition source.  The `source` only helps provide the location of documentation about the origins and details of the units specified for the reader, but it will not be used or checked by EdgeX.  An optional default source can be provided at the top level configuration (as shown in the examples below) so that other sources are only need when there specific units used that are not found in the default source.
+
+The units of measure can be categorized for better organization and to allow for different sources to be specified for different units.  The categories are defined by the TOML section names (the UoM dot labels).
+
+**Sample TOML unit of measure configuration**
+
+``` toml
+[Uom]
+    source="reference to source for all UoM if not specified below"
+    [Uom.temperature]
+    source="www.weather.com"
+    values="C,F,K"
+    [Uom.weights]
+    source="www.usa.gov/federal-agencies/weights-and-measures-division"
+    values="lbs,ounces,kilos,grams"
+```
+
+!!! Note
+        In the future, the source may be used by EdgeX to dynamically pull in units of measure rather than explicitly listing them in a configuration file.  As there is no standard for this type of data exchange today, this goes beyond the complexity of EdgeX needs, and would be left to the adopter to implement this at their discretion if desired.
```

Comment:
Isn't this already made possible by the UoMFile field accepting URIs?

Reply:
Yes - this was an old comment before the new approach of using an external file.  I pulled it out.  Thanks Farshid.